be1-go/hub/base_hub.go

```diff
@@ -57,7 +68,9 @@ func (h *baseHub) Start() {
 		for {
 			select {
 			case incomingMessage := <-h.messageChan:
-				h.handleIncomingMessage(&incomingMessage)
+				h.workers.Acquire(context.Background(), 1)
+				h.workersWg.Add(1)
+				go h.handleIncomingMessage(&incomingMessage)
```

Comment:
I'm not 100% convinced by this, but at least it provides some feedback on a potential overloading of the system:

```go
				ok := h.workers.TryAcquire(1)
				if !ok {
					log.Print("warn: worker pool full, waiting...")
					h.workers.Acquire(context.Background(), 1)
				}

				h.workersWg.Add(1)
				go h.handleIncomingMessage(&incomingMessage)
```